src/plugins/browserLogsPlugin.js

```diff
@@ -1,39 +1,44 @@
-/* eslint no-console: */
-
 var _ = require('lodash');
+var logging = require('selenium-webdriver').logging;
+
+var DEFAULT_LOG_LEVEL = logging.Level.OFF;
+
 function BrowserLogsPlugin(config, instanceConfig, logger) {
-  this.level = _.get(config, 'log.browser.level', 'error');
+  this.config = config;
   this.logger = logger;
-}
+  this.browserLoggingPref = _.get(config, 'log.browser.level', DEFAULT_LOG_LEVEL.name).toUpperCase();
 
-BrowserLogsPlugin.prototype.suiteStarted = function () {
-  var that = this;
-  if (that.level) {
-    browser.executeScriptHandleErrors('startLogCollection', {level: that.level})
-      .catch(function () {
-        // swallow error, already logged on debug level, avoid double logs
-      });
+  var browserLevel = logging.Level[this.browserLoggingPref];
+  if (browserLevel && browserLevel.value < DEFAULT_LOG_LEVEL.value) {
+    this.active = true;
   }
-};
+}
 
 BrowserLogsPlugin.prototype.specDone = function () {
-  browser.executeScriptHandleErrors('getAndClearLogs')
-    .then(function (logs) {
-      var template = _.template('BROWSER LOG: ${level}: ${message}');
-      _.each(logs, function (log) {
-        console.log(template(log));
-      });
-    })
-    .catch(function () {
-      // swallow error, already logged on debug level, avoid double logs
-    });
+  return this._logBrowserMessages();
+};
+
+BrowserLogsPlugin.prototype.onRuntimeSetup = function (runtime) {
```

Comment:
The approach is fine but I want to avoid the coupling to the concept of runtime. What about a method called onConnectionSetup that is called from DriverConnectionProvider.getNewDriver() with the resolved capabilities object only ? (jus before the logging on line 409 ?